### PR TITLE
Story page content type migration

### PR DESCRIPTION
**Pull request by weerd**
### What does this PR do?

This PR add a new content-type generated migration script for the new `StoryPage` content type.

### Any background context you want to provide?

I refrained from including a `blocks` field for the time being to keep things simple. I foresee likely needing to add it to this page type in the future.

I also limited the RichText field options to only markdown for now. It does not include support for Hyperlinks or Emebedded entries and assets for now. This is only temporary as I build out more robust support for the RichText field and how its content is parsed.

### What are the relevant tickets/cards?

Refs [Pivotal ID #163913064](https://www.pivotaltracker.com/story/show/163913064)

### CODE REVIEW

File: contentful/content-types/storyPage.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        unique: true,
+      },
+      {
+        size: {
+          max: 100,
+        },
+      },
+      {
+        regexp: {
+          pattern: '^(?!/)story/[a-zA-Z0-9-/]+$',
```
Comment (weerd) [inline contentful/content-types/storyPage.js]: Made it so that the slug has to begin with `story/` 💃

File: contentful/content-types/storyPage.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    .omitted(false)
+    .linkType('Entry');
+
+  storyPage
+    .createField('content')
+    .name('Content')
+    .type('RichText')
+    .localized(false)
+    .required(false)
+    .validations([
```
Comment (mendelB) [inline contentful/content-types/storyPage.js]: Why do we have these validations here unlike in CompanyPages? Are we excluding certain Markdown types?

File: contentful/content-types/storyPage.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    helpText:
+      'This title is used internally to help find this content. It will not be displayed anywhere on the rendered web page.',
+  });
+
+  storyPage.changeEditorInterface('title', 'singleLine', {});
+  storyPage.changeEditorInterface('subTitle', 'singleLine', {});
+
+  storyPage.changeEditorInterface('slug', 'slugEditor', {
+    helpText:
+      'If it applies, add an appropriate category prefix for the slug, e.g. "articles/", "about/", "facts/", etc.',
```
Comment (mendelB) [inline contentful/content-types/storyPage.js]: Should we update this to be just `"story/"`?

File: contentful/content-types/storyPage.js (hunk 2 above)
Comment (weerd) [reply contentful/content-types/storyPage.js]: Yeah, I mention it in description. I'm keeping it to only simple markdown for now on these pages until I can further enhance how we handle the RichText field on Phoenix :)

File: contentful/content-types/storyPage.js (hunk 3 above)
Comment (weerd) [reply contentful/content-types/storyPage.js]: Oh ya.

File: contentful/content-types/storyPage.js (hunk 2 above)
Comment (mendelB) [reply contentful/content-types/storyPage.js]: Ohhh ok, I didn't realize that this is what's needed to invalidate embedded links

Comment (weerd) [reply contentful/content-types/storyPage.js]: Yeah, this is all just temporary. As this is all getting built out 💃

**Pull request merged**